Profile/Profile/Presentation/DatesAndCalendar/CoursesToSyncView.swift

```diff
@@ -0,0 +1,170 @@
+                    .ignoresSafeArea(.all, edges: .bottom)
+                }
+                .navigationBarHidden(true)
+                .navigationBarBackButtonHidden(true)
+                
+                if viewModel.showError {
+                    errorAlertView
+                }
+            }
+            .ignoresSafeArea(.all, edges: .horizontal)
+        }
+    }
+    
+    // MARK: - Navigation and Title
+    private var navigationAndTitle: some View {
+        ZStack {
+            HStack {
+                Text(ProfileLocalization.CoursesToSync.title)
+                    .titleSettings(color: Theme.Colors.loginNavigationText)
+                    .accessibilityIdentifier("courses_to_sync_text")
+            }
+            VStack {
+                Button(
+                    action: { viewModel.router.back() },
+                    label: {
+                        CoreAssets.arrowLeft.swiftUIImage.renderingMode(.template)
+                            .backButtonStyle(color: Theme.Colors.loginNavigationText)
+                    }
+                )
+                .foregroundColor(Theme.Colors.styledButtonText)
+                .padding(.leading, isHorizontal ? 48 : 0)
+                .accessibilityIdentifier("back_button")
+                
+            }.frame(minWidth: 0,
+                    maxWidth: .infinity,
+                    alignment: .topLeading)
+        }
+    }
+    
+    private var coursesList: some View {
+        VStack(alignment: .leading, spacing: 24) {
+            ForEach(
+                Array(
+                    viewModel.coursesForSync.filter({ course in
+                        course.synced == viewModel.synced && (!viewModel.hideInactiveCourses || course.active)
+                    }).enumerated()
+                ),
+                id: \.offset
+            ) { _, course in
+                HStack {
+                    CheckBoxView(
+                        checked: Binding(
+                            get: { course.synced },
+                            set: { _ in viewModel.toggleSync(for: course) }
+                        ),
+                        text: course.name,
+                        color: Theme.Colors.textPrimary.opacity(course.active ? 1 : 0.8)
+                    )
+                    
+                    if !course.active {
+                        Text(ProfileLocalization.CoursesToSync.inactive)
+                            .font(Theme.Fonts.labelSmall)
+                            .foregroundStyle(Theme.Colors.textPrimary.opacity(course.active ? 1 : 0.8))
+                    }
+                }
+                .frame(
+                    minWidth: 0,
+                    maxWidth: .infinity,
+                    alignment: .leading
+                )
+            }
+        }
+        .padding(.horizontal, 24)
+        .padding(.vertical, 16)
+
+    }
+    
+    // MARK: - Error Alert View
+    private var errorAlertView: some View {
```

Comment:
Also, all views that use ```SnackBarView(message: viewModel.errorMessage)``` must be unified in all project in separate brach.

Reply:
@IvanStepanok 
I agree on unifying alerts, and it would be great to attach the created ticket here (to keep track of this task on our board).
My question is why copy-paste the same code three times (even if we want to redo the alerts in the project). Wouldn't it be better to create a shared view once and reuse it in all the necessary places? We could interpret this view as the beginning of unification. WDYT?